IS-66 - Create the Virtual Classrooms program view

crayolakat opened the pull request.

crayolakat committed:
Changed to faded photo

crayolakat committed:
Changed to faded photo

crayolakat committed:
Added VirtualClasssroomsPanel

crayolakat committed:
Added VirtualClasssroomsPanel

crayolakat committed:
Added svg icons to Virtual Classrooms page

crayolakat committed:
Added svg icons to Virtual Classrooms page

crayolakat committed:
Modified descriptions for Virtual Classrooms icons

crayolakat committed:
Modified descriptions for Virtual Classrooms icons

crayolakat committed:
Enlarged Virtual Classrooms icons

crayolakat committed:
Enlarged Virtual Classrooms icons

crayolakat committed:
Added Program Footer component

crayolakat committed:
Added Program Footer component

crayolakat committed:
Increased height of Program Footer

crayolakat committed:
Increased height of Program Footer

bibekg commented:
Glad to see progress! Make sure you change the base branch on this PR to `programs-page-redesign` since we want to aggregate these changes there first. 

Also, did you create this branch off of `master` or that branch?

crayolakat commented:
I created this branch off the programs-page-redesign branch

bibekg commented:
Okay cool, in that case, just change the base branch for the PR, as described above.

crayolakat committed:
Merge branch 'IS-66' of https://github.com/the-coding-school/the-cs into IS-66

crayolakat commented:
Okay, I've changed the base.

bibekg commented:
Just took a look at it in-browser. Looks great! Only thing I'd suggest changing up is reducing the upper/lower margin on the text and adding a little bit of padding on the sides to make it more closely match with the mockup.
![image](https://user-images.githubusercontent.com/8083680/32187945-a5e16cb6-bd63-11e7-969c-81ef378dbf35.png)

crayolakat committed:
Added side padding and decreased top/bottom margin

Added padding to sides of Virtual Classroom footer text and increased line height of text to decrease top and bottom margin.

crayolakat commented:
I added padding to the sides and increased the line height to reduce the upper/lower margins. I think the upper/lower margins still look a bit too big, do you think I should increase the text size or crop the picture?

bibekg commented:
I think you should decrease the text size a little bit (maybe to `$text_sm` or `$text_md`) and then reduce your `$program_footer_height` SCSS variable to about 500px.

crayolakat committed:
Modified Program Footer

Made text size and height of photo smaller

crayolakat commented:
Made the changes, what do you think now?

crayolakat committed:
Fixed Internet Explorer Issue

Added polyfill for Array.prototype.find method

crayolakat committed:
Fixed Internet Explorer text wrap issue

crayolakat commented:
I also added a few things to the code to make it compatible with Internet Explorer.

bibekg commented on src/views/Navbar/NavbarDropdown/NavbarDropdown.js:
```diff
@@ -57,6 +57,50 @@ export default function NavbarDropdown(props) {
 
   // Looks into the 'sections' property exported by each page component to see
   // what anchor-links that page makes available
+  //Polyfill for Array.prototype.find (method is not compatible with Internet Explorer)
```
Nice, great work finding the IE bug!

Only comment here is, polyfills should be not be placed within a React component. A good idea would be to move it to `src/utilities/polyfill.js`, then just throw in a 
```js
import './utilities/polyfill'
```
at the top of `src/index.js`

bibekg reviewed (changes requested):
Left a comment on a the polyfill. Also, I'd like to suggest a few other changes:
1. Move the ProgramFooter component from `components` to `pages/OurProgramsPage/ProgramViews` since it's not really going to be used anywhere besides ProgramViews.
2. Same thing goes for the `views/VirutalClassroom` component. It should just be located within `pages/OurProgramsPage/VirtualClassroomsView`
3. I noticed there are two instances of the footer image in this pull request as both `footer.jpg` and `virtual-learning.jpg`. Delete one and resize the other so it has a width of ~1000px in order to reduce file size/page load time.

crayolakat committed:
Moved polyfill out of React component

crayolakat committed:
Moved ProgramFooter component

crayolakat committed:
Moved VirtualClassroom components

crayolakat committed:
Resized and removed duplicate VirtualClassrooms footer pic

crayolakat committed:
Fixed IE bug

Bug caused program options to extend off right of screen

crayolakat commented:
I made the changes and fixed another IE bug that I noticed. I actually noticed a lot of IE bugs, but have not had the time to fix. Do you want me to start logging them somewhere?

bibekg commented:
That would be great. Just file them as issues here on the Github.

bibekg commented:
Good work. 👍 I've assigned the Learning Disabilities program view to you to work on next.
